### PR TITLE
More specific name for view sorts

### PR DESCRIPTION
**Pull request by Leksat**
The "name" key is used as a base name for creating the name of the enum type. So it should be pretty specific.

### CODE REVIEW

**Comment by fubhy** [comment]
Let's instead change the logic for generating the enum type names so it does not use the field name but something else... like an additional array value.

**Review by fubhy — CHANGES_REQUESTED** [review]
See my other comment.

**Comment by Leksat** [comment]
@fubhy, something like this?

File: modules/graphql_core/src/GraphQL/Traits/TypedPluginTrait.php [hunk 1]
```diff
@@ -90,7 +90,10 @@ protected function buildType(GraphQLSchemaManagerInterface $schemaManager) {
         $type = array_pop($types) ?: $schemaManager->findByName('String', [GRAPHQL_CORE_SCALAR_PLUGIN]);
       }
       else if (array_key_exists('type', $definition) && $definition['type']) {
-        $type = is_array($definition['type']) ? $this->buildEnumConfig($definition['type'], $definition['name']) : $schemaManager->findByName($definition['type'], [
+        if (is_array($definition['type']) && !isset($definition['enum_type_name'])) {
+          var_dump($definition);
```
Comment (fubhy) [inline modules/graphql_core/src/GraphQL/Traits/TypedPluginTrait.php]: Whoops.

**Review by fubhy — CHANGES_REQUESTED** [review]

File: modules/graphql_core/src/GraphQLSchemaManager.php [hunk 2]
```diff
@@ -54,6 +54,9 @@ protected function getDefinitions() {
   public function find(callable $selector, array $types, $invert = FALSE) {
     $instances = [];
     foreach ($this->getDefinitions() as $index => $def) {
+      if (!isset($def['definition']['name'])) {
+        var_dump($def['definition']);
```
Comment (fubhy) [inline modules/graphql_core/src/GraphQLSchemaManager.php]: Whoops.

File: modules/graphql_views/src/Plugin/Deriver/ViewDeriver.php [hunk 3]
```diff
@@ -82,15 +82,15 @@ public function getDerivativeDefinitions($basePluginDefinition) {
...
+            "enum_type_name" => "ViewsSortDirectionEnum",
             "type" => [
               "ASC" => "Ascending",
               "DESC" => "Descending",
             ],
             "default" => TRUE,
           ],
           'sortBy' => [
-            "name" => "sortBy",
+            "enum_type_name" => graphql_core_camelcase([$id, 'SortByEnum']),
```
Comment (fubhy) [inline modules/graphql_views/src/Plugin/Deriver/ViewDeriver.php]: I think this should be SortBy{xxx}Enum

**Comment by fubhy** [comment]
@Leksat Yes, along those lines. But pleas without the var_dump() :D

@pmelab what do you think?

**Comment by Leksat** [comment]
Whoops, indeed :D

**Pull request merged**